cmake/SetupCompilerFlags.cmake

```diff
@@ -32,21 +32,6 @@ else ()
   message(STATUS "std::filesystem NOT found, using POSIX")
 endif ()
 
-check_cxx_source_compiles(
-  "#include <sanitizer/asan_interface.h>
-
-  int main(int, char**)
-  {
-    return 0;
-  }"
-  UMPIRE_HAS_ASAN)
-
-if (UMPIRE_HAS_ASAN)
-  message(STATUS "Umpire may be built with ASAN support")
```

Comment:
Are you sure we want to get rid of this? It seems like it could be good to have the option for automatic detection.. what if we modified to so that it will turn `UMPIRE_HAS_ASAN` to ON unless `ENABLE_ASAN` has been manually turned off?

Reply:
It's a good question. My inclination is to remove it as it is not reliable.  We have found (too many) systems where the header file exists, but the runtime support is not there.  Or the runtime advertises support, but the header file is missing.

Reply:
Oh I see. Well that makes sense. So I guess this means that ASAN support will be up to the user to figure out - which should be fine, right?